src/ProgressOnderwijsUtils/Collections/MaybeExtensions.cs

```diff
@@ -71,6 +73,10 @@ public static class MaybeExtensions
             }
         }
 
+        [Obsolete("When the state of a Maybe is a nested Maybe, the error of that nested state can easily be ignored.", true)]
+        public static Maybe<Maybe<TOkOut, TIgnoredError>, TError> WhenOk<TOk, TError, TOkOut, TIgnoredError>(this Maybe<TOk, TError> state, Func<TOk, Maybe<TOkOut, TIgnoredError>> map)

```

Comment:
Is dit niet beter een warning?  Want het is niet ondenkbaar dat je dit wilt toestaan (en warnings kun je onderdrukken), en we hebben toch warnings-as-error.

Reply:
Maar goed, als iets een echte error is zou ik een exceptie ipv implementatie overwegen, is ook duidelijker niet erg om niet te testen, of desnoods doorsluizen naar de oorsprongelijke method (kan dat eigenlik ueberhaupt makkelijk?)

Reply:
https://github.com/progressonderwijs/ProgressOnderwijsUtils/pull/445